toml/src/main/kotlin/org/rust/toml/crates/local/CrateVersion.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.crates.local
+
+import com.vdurmont.semver4j.Requirement
+import com.vdurmont.semver4j.Semver
+import com.vdurmont.semver4j.SemverException
+
+class CrateVersionRequirement private constructor(private val requirements: List<Requirement>) {
+    fun matches(version: Semver): Boolean = requirements.all {
+        it.isSatisfiedBy(version)
+    }
+
+    companion object {
+        fun build(text: String): CrateVersionRequirement? {
+            val requirements = text.split(",").map { it.trim() }
+            val parsed = requirements.mapNotNull {
+                try {
+                    Requirement.buildNPM(normalizeVersion(it))
+                } catch (e: SemverException) {
+                    return@mapNotNull null
+                }
+            }
+            if (parsed.size != requirements.size) return null
+
+            return CrateVersionRequirement(parsed)
+        }
+    }
+}
+
+/**
+ * Turns 1.2.3 into ^1.2.3.
+ * See https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio
+ */
+private fun normalizeVersion(version: String): String {
+    var normalized = version
+    if (normalized.firstOrNull() in listOf('<', '>', '=')) {
```

Comment:
Let's add some comments to describe what's going here and why it is necessary. It'd be great to have two comments, about `.0` and `^` respectively